controller/engine_controller.go

```diff
@@ -1393,6 +1398,40 @@ func (ec *EngineController) UpgradeEngineProcess(e *longhorn.Engine) error {
 	return nil
 }
 
-func (ec *EngineController) isResponsibleFor(e *longhorn.Engine) bool {
-	return isControllerResponsibleFor(ec.controllerID, ec.ds, e.Name, e.Spec.NodeID, e.Status.OwnerID)
+func (ec *EngineController) isResponsibleFor(e *longhorn.Engine, defaultEngineImage string) bool {
```

Comment:
see comment about other isResponsible function and see if the same issues apply here.

Reply:
Thank you. Fixed